src/targets/github.ts

```diff
@@ -378,12 +378,30 @@ export class GithubTarget extends BaseTarget {
         );
       }
 
-      // XXX: This is a bit hacky as we rely on two things:
-      // 1. GitHub issuing a redirect to S3, where they store the artifacts,
-      //    or at least pass those request headers unmodified to us
-      // 2. AWS S3 using the MD5 hash of the file for its ETag cache header
-      //    when we issue a HEAD request.
-      const response = await this.github.request(`HEAD ${url}`, {
+      const remoteChecksum = await this.getRemoteChecksum(url);
+      const localChecksum = createHash('md5').update(file).digest('hex');
```

Comment:
Should we preserve the note added in #323 about this local computation only working for simple cases?

Reply:
Nevermind. Now that we're computing the remote checksum locally (😜) we are guaranteed the same algorithm 👍

Reply:
Though in that case should we factor out a function that we call here and also beneath `md5FromUrl` to reinforce/ensure the same algorithm?

Reply:
If ETag is computed different for large files, as https://github.com/getsentry/craft/issues/322#issuecomment-964303174 suggests, that would still be a problem -- not something we're fixing here.

Do we have any repo with a release asset >> 1 GB?

I'm okay adding one more caveat to the list of assumptions the code makes.

As for "reinforce/ensure the same algorithm" -- do you mean wrapping `createHash('md5').update(BYTES).digest('hex')` as a function, or something else?

Reply:
> If ETag is computed different for large files

It is by S3, but we're not depending on S3's calculation anymore, we're downloading the entire asset and computing a hash ourselves.

> do you mean wrapping

Yeah that's all, a small helper function to guarantee that `remoteChecksum` and `localChecksum` remain identical. Wdyt?

Reply:
> we're downloading the entire asset and computing a hash ourselves.

We only do that for the case where the `ETag` header is not present. If it is present and different (because it was computed in a different way or any other reason) then we'll fail the comparison and the GitHub target will fail (as it does today).

By the time we reach that size concern I wish GitHub will have a proper API to get a checksum so we don't need to upload/download every asset 😄 

BTW we've also noticed #308 added both an asset size check and a checksum check. Most likely, the size check was enough to detect the sort of problem #308 was dealing with. If we run into more problems with ETag/MD5 and if re-downloading assets causes collateral damage, I'd consider removing the checksum altogether.

Reply:
> Yeah that's all, a small helper function to guarantee that `remoteChecksum` and `localChecksum` remain identical. Wdyt?

👍

Reply:
> We only do that for the case where the ETag header is not present.

Ah, good point! In that case I'll bring back my original suggestion:

> Should we preserve the note added in #323 about this local computation only working for simple cases?

☺️

Reply:
This has been resolved, added https://github.com/getsentry/craft/pull/328#issuecomment-971638002 to summarize the state.